regions_www/m/server.R

```diff
@@ -79,6 +78,17 @@ lsoa_legend_df <- data.frame(
   labels = c("1-9", "10-49", "50-99", "100-249",
              "250-499", "500-999", "1000-1999", "2000+")
 )
+oldEncodeShinySaveState <- getFromNamespace("encodeShinySaveState", "shiny")
+
+shortEncodeShinySaveState <- function(state){
+  res <- oldEncodeShinySaveState(state)
+  r <- httr::POST(
+    "https://www.googleapis.com/urlshortener/v1/url?key=AIzaSyBLGLnQojQm2WQu9Urri2mmVTQolrLuJBc",
```

Comment:
I generally suggest saving API keys in the `.Renviron` file, as described here: https://csgillespie.github.io/efficientR/set-up.html#renviron in this case. Is there no issue with rate limits? Maybe no issue sharing this with the world in that case... Another compromise, in this case between the + of short urls and the + of reducing dependencies. In this case I'd lean slightly towards reducing dependencies although can understand the desire to keep URLs short.

Reply:
Nudge on this @nikolai-b - suggest not publicising the key in the code.

Reply:
Thanks for spotting @Robinlovelace !  Very late response sorry I just saw this.  You can see the commit message is

> Dummy attempt at keeping URL short
> 
> Appears wrong on link, exposes our API key

This API key was locked down to only work from pct.bike and I disabled a few days after posting it here (was just so others could try it) but it just shows what we could do...
Since then the whole goo shortner has been [deprecated](https://developers.googleblog.com/2018/03/transitioning-google-url-shortener.html) so to be completely clear this is not the approach I'd suggest it was just to show it was possible to hack the URL to make it shorter, I think using `assignInNamespace` is probably a bad idea!

Reply:
Never come across `assignInMyNamespace`. The standard way for users to handle API keys now is:

```r
usethis::edit_r_environ()
# add new line like KEY=XYZ
# Restart R
Sys.getenv("KEY")
```

Would that approach work on the server (if we ever did want to use API keys)?

Late response to a late response!